modules/users/client/services/authentication.client.service.js

```diff
@@ -1,12 +1,35 @@
-'use strict';
+(function() {
+  'use strict';
 
-// Authentication service for user variables
-angular.module('users').factory('Authentication', ['$window',
-  function ($window) {
-    var auth = {
-      user: $window.user
+  angular
+    .module('users')
+    .service('Authentication', Authentication);
+
+  Authentication.$inject = ['$resource', '$q'];
+  function Authentication($resource, $q) {
+
+
+    var service = {
+      ready: init(),
+      user: null,
+      refresh: init
     };
 
-    return auth;
+    function init() {
+      return $q(function(resolve, reject) {
+        $resource('api/users/me')
```

Comment:
@trainerbill Can we add this api route to the Users service?

I'd rather see this service more focused with methods that pertain specifically to Authentication; logout, signin, and eventually any logic we need for Token-based authentication (JWT).